Toolset/libraries/revidelibrary.8.livecodescript

```diff
@@ -3874,13 +3874,21 @@ function revIDEExtensionFolders
    return tFoldersA
 end revIDEExtensionFolders
 
-function revIDEDefaultExtensionIcon pType
+function revIDEDefaultExtensionIcon pType, pRetina
    switch pType
       case "library"
-         return revIDESpecialFolderPath("images") & slash & "extension-library.png"
+         if pRetina is true then
+            return revIDESpecialFolderPath("images") & slash & "extension-library@extra-high.png"
+         else
+            return revIDESpecialFolderPath("images") & slash & "extension-library.png"
+         end if
       case "widget"
       default
-         return revIDESpecialFolderPath("images") & slash & "extension-widget.png"
+         if pRetina is true then
+            return revIDESpecialFolderPath("images") & slash & "extension-widget@extra-high.png"
+         else
+            return revIDESpecialFolderPath("images") & slash & "extension-widget.png"
+         end if
    end switch
```

Comment:
Well, it's also there to provide the paths for the icon and retina icon when packaging the widget, so from that point of view it's correct.

In the case of the extension builder plugin it's taking the binary data from the file and settings the imagedata to it, so it won't be affected by the engine's retina stuff. We could set the image path instead, but isn't the idea to display the icons that have been selected as the icon and retina icon, and therefore display them as they actually are? Otherwise they would both be retina on retina machines, and both non-retina on non-retina. That's fine if that's what we want, I'm not quite sure.